Regression/Core/src/main/java/org/tribuo/regression/ImmutableRegressionInfo.java

```diff
@@ -58,37 +63,67 @@ private ImmutableRegressionInfo(ImmutableRegressionInfo info) {
     }
 
+    /**
+     * Builds an ImmutableRegressionInfo from the supplied RegressionInfo.
+     * <p>
+     * Enforces that the ids are generated in lexicographic order.
+     * @param info The info to build from.
+     */
     ImmutableRegressionInfo(RegressionInfo info) {
         super(info);
         idLabelMap = new LinkedHashMap<>();
         labelIDMap = new LinkedHashMap<>();
+        // Sort count map
+        SortedSet<String> names = new TreeSet<>(countMap.keySet());
         int counter = 0;
-        for (Map.Entry<String,MutableLong> e : countMap.entrySet()) {
-            idLabelMap.put(counter,e.getKey());
-            labelIDMap.put(e.getKey(),counter);
+        for (String e : names) {
+            idLabelMap.put(counter,e);
+            labelIDMap.put(e,counter);
             counter++;
         }
         domain = calculateDomain(minMap);
     }
 
+    /**
+     * Builds an ImmutableRegressionInfo from the supplied RegressionInfo using the
+     * supplied id mapping.
+     * <p>
+     * This can result in id mappings which are not in lexicographic order.
+     * <p>
+     * Expects that the mapping is made up of single dimension regressors, usually
+     * {@link org.tribuo.regression.Regressor.DimensionTuple}, though it only enforces
+     * that a single dimension is present.
+     * @param info The info to build from.
+     * @param mapping The id number - dimension name bijective mapping.
+     */
     ImmutableRegressionInfo(RegressionInfo info, Map<Regressor,Integer> mapping) {
         super(info);
         if (mapping.size() != info.size()) {
             throw new IllegalStateException("Mapping and info come from different sources, mapping.size() = " + mapping.size() + ", info.size() = " + info.size());
         }
 
-        idLabelMap = new HashMap<>();
-        labelIDMap = new HashMap<>();
+        String[] names = new String[mapping.size()];
         for (Map.Entry<Regressor, Integer> e : mapping.entrySet()) {
             Regressor r = e.getKey();
-            String[] names = r.getNames();
-            if (names.length == 1) {
-                idLabelMap.put(e.getValue(), names[0]);
-                labelIDMap.put(names[0], e.getValue());
+            String[] curNames = r.getNames();
+            if (names[e.getValue()] != null) {
+                throw new IllegalArgumentException("Mapping must be a bijection, but found two mappings for index " + e.getValue() + ", '" + names[e.getValue()] + "' and '" + curNames[0] + "'");
+            }
+            else if (curNames.length == 1) {
+                names[e.getValue()] = curNames[0];
             } else {
                 throw new IllegalArgumentException("Mapping must contain a single regression dimension per id, but contains " + Arrays.toString(names) + " -> " + e.getValue());
             }
         }
+        idLabelMap = new LinkedHashMap<>();
+        labelIDMap = new LinkedHashMap<>();
+        for (int i = 0; i < names.length; i++) {
```

Comment:
Seems like 'names' should be sorted here but I don't think that it is.

Reply:
No, this is the intentional escape hatch, used when the output info needs to be constructed for an external model (e.g. loading in an ONNX or TF model) which doesn't have the same invariants as Tribuo requires.

Reply:
ok - I see that in the javadoc now.  thanks for the clarification.